### PR TITLE
Implement complex TSMT inert for depth 16->32 case

### PR DESCRIPTION
**Pull request by bobbinth**
This PR implements an insert procedure in std::collections::smt to handle the case when an inserted key-value pair collides with an existing leaf at depth 16 and both key-value pairs must be moved to leaves at depth 32.

### CODE REVIEW

**Review by frisitano — APPROVED** [review]
Looks good!

File: stdlib/asm/collections/smt.masm [hunk 1]
```diff
@@ -605,6 +605,133 @@ proc.insert_48.2
...
+    # drop the E32 root as we don't need it, and arrange the stack for inserting the next
+    # leaf (12 cycles)
+    dropw swapw dropw swapw swapw.3 swapw.2
+    # => [X, N, T, P, R, ...]
+
+    # load the indexes from loc[0] and drop all but idx_lo from the stack (7 cycles)
+    loc_loadw.0 drop drop swap drop
+    # => [idx_lo, N, T, P, R, ...]
+
+    # insert node M into the subtree with root T at depth 16; this leaves the new root of the
```
Comment (Al-Kindi-0) [inline stdlib/asm/collections/smt.masm:710]: nit: node M -> node N

**Review by Al-Kindi-0 — APPROVED** [review]
Looks great, @bobbinth ! The procedure turned out to be quiet involved but the inlined docs greatly help to lessen the complexity of the code. I personally don't see any immediate simplifications or optimizations.

File: processor/src/decorators/adv_stack_injectors.rs [hunk 2]
```diff
@@ -428,6 +430,25 @@ where
                 // return is_update = ONE, is_simple_insert = ZERO
                 (ONE, ZERO)
             } else {
+                // TODO: improve code readability as more cases are handled
```
[the remark this answers is not in the document]
Comment (bobbinth) [reply processor/src/decorators/adv_stack_injectors.rs:433]: Agreed! Will implement in the next PR.

**Pull request merged**